### PR TITLE
[docs] Add plan in the nav bar for pro-only and premium-only pages

### LINKED ISSUE #4581: [docs] Add plan icon for pro-only and premium-only pages on navigation bar
Since #4441, we have pro-only and premium-only doc pages
They should be clearly marked as such with the pro and premium icon on the navigation bar.

### PR DESCRIPTION
**Pull request by flaviendelangle**
Closes #4581

image

Preview: https://deploy-preview-4591--material-ui-x.netlify.app/x/react-data-grid/

### CODE REVIEW

File: docs/data/pages.ts [hunk 1]
```diff
@@ -86,7 +88,7 @@ const pages = [
       { pathname: '/x/react-date-pickers/time-picker' },
       {
         pathname: '/x/api/date-pickers',
-        title: ' • API Reference',
+        title: 'API Reference',
```
Comment (flaviendelangle) [inline docs/data/pages.ts:91]: I prefer without it but I can add it back if you want

File: docs/pages/_app.js [hunk 2]
```diff
@@ -256,7 +255,7 @@ function AppWrapper(props) {
           <PageContext.Provider value={{ activePage, pages: productPages }}>
             <ThemeProvider>
               <DocsStyledEngineProvider cacheLtr={emotionCache}>
-                <XWrapper>{children}</XWrapper>
+                {children}
```
Comment (flaviendelangle) [inline docs/pages/_app.js:258]: The pro / premium icon CSS is now defined on the core

**Review by m4theushw — APPROVED** [review]
The "Date range picker" also needs the Pro icon.

image

**Comment by oliviertassinari** [comment]
We have an issue with the side nav. on https://master--material-ui-x.netlify.app/x/react-data-grid/tree-data/ the navigation is closed

**Actual**
Screenshot 2022-04-23 at 19 07 55

**Expected**
Screenshot 2022-04-23 at 19 07 58

**Pull request merged**